sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -659,13 +660,19 @@ case class StructsToJson(
         (arr: Any) =>
           gen.write(arr.asInstanceOf[ArrayData])
           getAndReset()
+      case MapType(_: DataType, _: StructType, _: Boolean) =>
+        (map: Any) =>
+          val mapType = child.dataType.asInstanceOf[MapType]
+          gen.write(map.asInstanceOf[MapData], mapType)
+          getAndReset()
     }
   }
 
   override def dataType: DataType = StringType
 
   override def checkInputDataTypes(): TypeCheckResult = child.dataType match {
-    case _: StructType | ArrayType(_: StructType, _) =>
+    case _: StructType | ArrayType(_: StructType, _) |
+         MapType(_: DataType, _: StructType, _: Boolean) =>
```

Comment:
I see. Looks now we have assumption that `JacksonGenerator` only takes `StructType` now. I think expression name should be fine and I guess we are okay to rename it if it looks required (actually I did it to `StructsToJson` :) ).

Would it be difficult to support `MapType` like `StructType`? I took a quick look and it looks quite a lot of places to fix though. Up to my understanding, we could load JSON both `MapType` and `StructType` too.

```
scala> spark.read.schema("a map<string, int>").json(Seq("""{"a": {"a": 2}}""").toDS).show()
+-----------+
|          a|
+-----------+
|Map(a -> 2)|
+-----------+


scala> spark.read.schema("a struct<a: int>").json(Seq("""{"a": {"a": 2}}""").toDS).show()
+---+
|  a|
+---+
|[2]|
+---+
```